docs/documentation.html

```diff
@@ -392,17 +395,33 @@ <h3 id="generating-a-parser-javascript-api">JavaScript API</h3>
   <dd>Callback for informational messages. See <a href="#error-reporting">Error Reporting</a></dd>
 
   <dt><code>output</code></dt>
-  <dd><p>If set to <code>"parser"</code>, the method will return generated parser
-  object; if set to <code>"source"</code>, it will return parser source code as
-  a string (default: <code>"parser"</code>).
-  If set to <code>"source-and-map"</code>, it will return a <a href="https://github.com/mozilla/source-map#sourcenode"><code>SourceNode</code></a> object; you can
-  get source code by calling <code>toString()</code> method or source code and mapping by
-  calling <code>toStringWithSourceMap()</code> method, see the <a href="https://github.com/mozilla/source-map#sourcenode"><code>SourceNode</code></a> documentation
-  (default: <code>"parser"</code>)</p>
-  <blockquote>
-  <p><strong>Note</strong>: because of bug <a href="https://github.com/mozilla/source-map/issues/444">source-map/444</a> you should also set <code>grammarSource</code> to
-  a not-empty string if you set this value to <code>"source-and-map"</code></p>
-  </blockquote></dd>
+  <dd><p>A string, one of:</p>
+    <ul>
+      <li><code>"parser"</code> - return generated parser object.</li>
+      <li><code>"source"</code> - return parser source code as a string.</li>
+      <li><code>"source-and-map"</code> - return a
+        <a href="https://github.com/mozilla/source-map#sourcenode"><code>SourceNode</code></a>
+        object; you can get source code by calling <code>toString()</code>
+        method or source code and mapping by calling
+        <code>toStringWithSourceMap()</code> method, see the
+        <a href="https://github.com/mozilla/source-map#sourcenode"><code>SourceNode</code></a>
+        documentation.
+      </li>
+      <li><code>"source-with-inline-map"</code> - return the parser source along
+        with an embedded source map as a <code>data:</code> URI.  This option
+        leads to a larger output string, but is the easiest to integrate with
+        developer tooling.</li>
+    </ul>
+    <p>(default: <code>"parser"</code>)</p>
+    <blockquote>
+      <p><strong>Note</strong>: because of bug <a
+      href="https://github.com/mozilla/source-map/issues/444">source-map/444</a>
+      you should also set <code>grammarSource</code> to a not-empty string if
+      you set this value to <code>"source-and-map"</code> or
+      <code>"source-with-inline-map"</code>.  The path should be relative to
+      the location where the generated parser code will be stored.</p>
```

Comment:
The last sentence seems not relevant for `source-with-inline-map`

Reply:
I think it is, actually.  If you generate with grammar source "src/foo.peggy", and write the output file to "lib/foo.peggy", then the grammar source needs to be "../src/foo.peggy".  On the CLI, you can use fully-qualified paths, because i use node's `path.relative` to fix them up.  Here, I don't want to pull in `path` or a polyfill, so I just change the requirement so that the grammarSource has to be relative.

Reply:
Never-mind, my mistake. I mistakenly thought that it is the path to the file into which the source map is embed, that is why I was doubt is it required to have a concrete value at all.

However, perhaps add your example to the doc?
I mean, is the concrete path matter for `source-with-inline-map` mode? Soure map is embedded to the

Reply:
An example is a great idea. Adding.